### ISSUE TITLE
oidcMiddleware prevents dispatch from returning

### ISSUE THREAD

**Issue by WolfspiritM** [maxmantz/redux-oidc#50]
Hi,

When using the oidcMiddleware in Version 3 (beta) it prevents dispatch from returning anything.
In our case we needed to get a Promise from the action but after implementing oidcMiddleware we only received undefined.

Lets say having following action:
``` js
function fetchAsync() {
 return function (getState, dispatch) {
   return fetch....
 }
}
```
using dispatch(fetchAsync()).then() will result in an error as the result from dispatch will be undefined.

This is fixed when changing:
https://github.com/maxmantz/redux-oidc/blob/master/src/oidcMiddleware.js#L46
to
``` js
  // the middleware
  return (store) => (next) => (action) => {
    return co(middlewareHandler(next, action, userManager));
  }
```

I'm however not sure if that will fix non Promise return types cause of "co" returning a promise instead.

**Comment by maxmantz** [comment]
This is indeed an issue with `co`. Since `co` doesn't allow promises to be returned, I've removed it from the dependencies and reverted back to using regular promises. These changes have been implemented in `v3.0.0-beta.8`.

**Comment by maxmantz** [comment]
I'll close this for now. Feel free to reopen or comment if this issue persists.

**Comment by sulfurious** [comment]
We started checking out redux-oidc for use in our project but are running into the issue above. We have a couple places where we are expecting a promise to be returned but instead are getting undefined.

We've fixed it by adding a return statement to return the call to middlewareHandler. Seems to make sense. Since, without it, any value returned from middlewareHandler would be stranded.

```
// the middleware
return (store) => (next) => (action) => {
  // added return below
  return middlewareHandler(next, action, userManager);
}
```
